src/NotifyMeFactory.php

```diff
@@ -43,12 +43,24 @@ public function createFactory(array $config)
             throw new InvalidArgumentException("A driver must be specified.");
         }
 
-        $class = "NotifyMeHQ\{$config['driver']}\{$config['driver']}Factory";
+        $class = $this->getDriverClassName($config['driver']);
 
         if (class_exists($class)) {
-        	return new $class();
+            return new $class();
         }
 
         throw new InvalidArgumentException("Unsupported driver [{$config['driver']}]");
     }
+
+    /**
+     * Resolve gateway class name from config.
+     *
+     * @param string $driver
+     *
+     * @return string
+     */
+    public function getDriverClassName($driver)
```

Comment:
This abstraction isn't required, and just adds an extra method call with no really benefit.